packages/@webex/internal-plugin-metrics/src/call-diagnostic/config.ts

```diff
@@ -6,7 +6,32 @@ import {ClientEventError} from '../metrics.types';
 
 export const NEW_LOCUS_ERROR_CLIENT_CODE = 4008;
 export const MEETING_INFO_LOOKUP_ERROR_CLIENT_CODE = 4100;
+export const UNKNOWN_ERROR = 9999; // Unexpected error that is not a meetingInfo error, locus error or browser media error.
 export const ICE_FAILURE_CLIENT_CODE = 2004;
+export const WBX_APP_API_URL = 'wbxappapi'; // MeetingInfo WebexAppApi response object normally contains a body.url that includes the string 'wbxappapi'
+
+// Found in https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
+const BROWSER_MEDIA_ERROR_NAMES = {
+  PERMISSION_DENIED_ERROR: 'PermissionDeniedError',
+  NOT_ALLOWED_ERROR: 'NotAllowedError',
+  NOT_READABLE_ERROR: 'NotReadableError',
+  ABORT_ERROR: 'AbortError',
+  NOT_FOUND_ERROR: 'NotFoundError',
+  OVERCONSTRAINED_ERROR: 'OverconstrainedError',
+  SECURITY_ERROR: 'SecurityError',
+  TYPE_ERROR: 'TypeError',
+};
+
+export const BROWSER_MEDIA_ERROR_NAME_TO_CLIENT_ERROR_CODES_MAP = {
```

Comment:
all these errors prevent us from using a camera or a microphone, but they are not fatal, the users will see in the UI that they can't unmute audio or video, but they can still join a meeting, so I think we need to make sure that the errors we map to are all the kind of errors that CA will treat as non-critical and will not mark the call as failed. 

I see some values mapped to 2001 - that's "FailedToConnectMedia" error, which doesn't feel right. The fact whether we have camera or mic doesn't affect if we can connect media or not. We will be able to connect "the rest" of the media - so for example users will still see remote video, share, hear audio etc - media connection will be successful.

The other value used: 2729 I guess it's fine, because I see it's in the "expected" category

Reply:
ohh oki, what about using CA's non-fatal error 'StreamErrorNoMedia'

https://sqbu-github.cisco.com/WebExSquared/event-dictionary/wiki/Error-codes-for-metric-events#:~:text=3013,false